python/spec/dependabot/python/update_checker/poetry_version_resolver_spec.rb

```diff
@@ -68,6 +69,22 @@
       it { is_expected.to eq(Gem::Version.new("2.18.4")) }
     end
 
+    context "with a dependency defined under dev-dependencies" do
+      let(:pyproject_content) do
+        super().gsub(/\[tool\.poetry\.dependencies\]/, "[tool.poetry.dev-dependencies]")
+      end
+
+      it { is_expected.to eq(Gem::Version.new("2.18.4")) }
+    end
+
+    context "with a dependency defined under a group" do
```

Comment:
From similar changes in the file parser at #6417 it would be useful to add a test for a non-dev group like `tool.poetry.group.docs.dependencies`.

Reply:
Sure, I can add that. Thanks for the feedback @phillipuniverse!